### PR TITLE
Add unique identifier to auth failure message

### PR DESCRIPTION
**Pull request by pontusmelke**

### CODE REVIEW

File: community/bolt/src/main/java/org/neo4j/bolt/security/auth/AuthUtils.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public static String uniqueIdentifier( StoreId storeId )
+    {
+        MessageDigest messageDigest;
+        try
+        {
+            messageDigest = MessageDigest.getInstance( "SHA-256" );
+            messageDigest.update( UTF8.encode( storeId.toString() ) );
+            byte[] digest = messageDigest.digest();
+            StringBuilder builder = new StringBuilder( digest.length * 2 );
+            for ( byte b : digest )
```
Comment (jakewins) [inline community/bolt/src/main/java/org/neo4j/bolt/security/auth/AuthUtils.java]: Prefer org.neo4j.kernel.impl.util.HexPrinter here

File: community/bolt/src/test/java/org/neo4j/bolt/v1/transport/integration/AuthenticationIT.java [hunk 2]
```diff
@@ -115,7 +116,7 @@ public void shouldFailIfWrongCredentials() throws Throwable
         // Then
         assertThat( client, eventuallyRecieves( new byte[]{0, 0, 0, 1} ) );
         assertThat( client, eventuallyRecieves( msgFailure( Status.Security.AuthenticationFailed,
-                "The client provided an incorrect username and/or password." ) ) );
+                String.format( "The client provided an incorrect username and/or password. (ID:%s)", server.uniqueIdentier()) ) ) );
```
Comment (jakewins) [inline community/bolt/src/test/java/org/neo4j/bolt/v1/transport/integration/AuthenticationIT.java:118]: This is perfect for what we need IMO, easy to parse out and human readable without committing ourselves to an API we don't want to support in the long term.

**Comment by jakewins** [comment]
Other than the comment about HexPrinter (which, since there are test failures, might as well change that) LGTM!

**Comment by pontusmelke** [comment]
@jakewins Fixed test failure and changed to use `HexPrinter`

**Pull request merged**